fix(curriculum): add `for` attribute to `label` elements in registration-form project (#47332)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md

---
id: 62ff919a7b5612c0670923a5
title: Step 37
challengeType: 0
dashedName: step-37
---

# --description--

> Now, to follow accessiblity best practices, link the form elements and their `label` elements together using the `for` attribute.

Reviewer: ```suggestion
Link the applicable form elements and their `label` elements together.
```

> Use these values in order for the `id`s: `profile-picture`, `age`, `referrer`, `bio`

Reviewer: ```suggestion
Use these values for the respective `id` properties: `profile-picture`, `age`, `referrer`, `bio`
```

# --hints--

The first `input` element should have an `id` of `profile-picture`.

```js

> assert(document.querySelectorAll('fieldset:nth-of-type(3) input')[0].matches('#profile-picture'))

Reviewer: ```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) input')?.[0]?.matches('#profile-picture'))
```

```

The second `input` element should have an `id` of `age`.

```js

> assert(document.querySelectorAll('fieldset:nth-of-type(3) input')[1].matches('#age'))

Reviewer: ```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) input')?.[1]?.matches('#age'))
```

```

The `select` element should have an `id` of `referrer`.

```js

> assert(document.querySelector('fieldset:nth-of-type(3) select').matches('#referrer'))

Reviewer: ```suggestion
assert(document.querySelector('fieldset:nth-of-type(3) select')?.matches('#referrer'))
```

```

The `textarea` element should have an `id` of `bio`.

```js

> assert(document.querySelector('fieldset:nth-of-type(3) textarea').matches('#bio'))

Reviewer: ```suggestion
assert(document.querySelector('fieldset:nth-of-type(3) textarea')?.matches('#bio'))
```

```

The first `label` element should have a `for` attribute with a value of `profile-picture`.

```js

> assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[0].matches('label[for="profile-picture"]'))

Reviewer: ```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[0]?.matches('label[for="profile-picture"]'))
```

```

The second `label` element should have a `for` attribute with a value of `age`.

```js

> assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[1].matches('label[for="age"]'))

Reviewer: ```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[1]?.matches('label[for="age"]'))
```

```

The third `label` element should have a `for` attribute with a value of `referrer`.

```js

> assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[2].matches('label[for="referrer"]'))

Reviewer: ```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[2]?.matches('label[for="referrer"]'))
```

```

The fourth `label` element should have a `for` attribute with a value of `bio`.

```js

> assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[3].matches('label[for="bio"]'))

Reviewer: ```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[3]?.matches('label[for="bio"]'))
```
